Modules/_struct.c

```diff
@@ -791,6 +822,58 @@ np_double(_structmodulestate *state, char *p, PyObject *v, const formatdef *f)
     return 0;
 }
 
+#ifdef Py_HAVE_C_COMPLEX
+static int
+np_float_complex(_structmodulestate *state, char *p, PyObject *v,
+                 const formatdef *f)
+{
+    Py_complex c = PyComplex_AsCComplex(v);
+    float complex x = CMPLXF((float)c.real, (float)c.imag);
+
+    if (c.real == -1 && PyErr_Occurred()) {
+        PyErr_SetString(state->StructError,
+                        "required argument is not a complex");
+        return -1;
+    }
+    memcpy(p, (char *)&x, sizeof(x));
```

Comment:
I'm not sure that the `(char *)` cast is useful. Same remark below.

Reply:
I suspect it's useless as well.  Yet, it's used in all helpers, e.g. np_double right above.

Maybe clean up these casts across the file?

Reply:
From the statistics, it seems that such remnants are mostly in this module:
```
$ git grep 'memcpy.*(char *\*)' | cut -f1 -d:|sort -n|uniq -c|sort -n
      1 Include/internal/pycore_call.h
      1 Modules/_ctypes/cfield.c
      1 Objects/call.c
      1 Objects/longobject.c
      1 Python/dtoa.c
      2 Modules/_io/bufferedio.c
      2 Objects/unicodeobject.c
      3 Modules/socketmodule.c
      5 Objects/memoryobject.c
     30 Modules/_struct.c
```